core/src/main/kotlin/com/strumenta/kolasu/parsing/KolasuParser.kt

```diff
@@ -74,8 +105,8 @@ abstract class KolasuANTLRLexer : KolasuLexer<KolasuANTLRToken> {
  *
  * You should extend this class to implement the parts that are specific to your language.
  */
-abstract class KolasuParser<R : Node, P : Parser, C : ParserRuleContext, T : KolasuToken> :
-    KolasuLexer<T>,
+abstract class KolasuParser<R : Node, P : Parser, C : ParserRuleContext, T : KolasuToken>(tokenFactory: TokenFactory<T>) :
+    KolasuANTLRLexer<T>(tokenFactory),
```

Comment:
Note that this was renamed in KolasuANTLRParser in 1.6, so it makes sense that here implements KolasuANTLRLexer and not KolasuLexer

Reply:
Should this get a default value (an instance of the DefaultTokenFactory)?

Reply:
It would be great if it had a default value, but it wouldn't fulfill its generics signature